service/src/test/java/skills/intTests/saml2/SAML2IntegrationIT.groovy

```diff
@@ -0,0 +1,206 @@
+/**
+ * Copyright 2024 SkillTree
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package skills.intTests.saml2
+
+import dasniko.testcontainers.keycloak.KeycloakContainer
+import org.keycloak.admin.client.Keycloak
+import org.openqa.selenium.WebDriver
+import org.openqa.selenium.By
+import org.openqa.selenium.htmlunit.HtmlUnitDriver
+import org.openqa.selenium.support.ui.WebDriverWait
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.boot.test.web.server.LocalServerPort
+import org.springframework.boot.test.web.client.TestRestTemplate
+import org.springframework.http.HttpMethod
+import org.springframework.http.HttpEntity
+import org.springframework.http.HttpHeaders
+import org.springframework.http.ResponseEntity
+import skills.SpringBootApp
+import org.springframework.http.HttpStatus
+import skills.auth.UserAuthService
+import skills.auth.form.CreateAccountController
+import spock.lang.IgnoreIf
+import spock.lang.Shared
+import spock.lang.Specification
+import org.openqa.selenium.Cookie
+import java.time.Duration
+
+@IgnoreIf({env["SPRING_PROFILES_ACTIVE"] == "pki" })
+@SpringBootTest(webEnvironment=SpringBootTest.WebEnvironment.RANDOM_PORT, classes = SpringBootApp)
```

Comment:
Please find attached suggested improvements for the integration test, zip file contains `SAML2IntegrationIT.groovy` and  `realm-export.json`
[SAML2IntegrationIT-updates.zip](https://github.com/user-attachments/files/17938224/SAML2IntegrationIT-updates.zip)

Updates to `SAML2IntegrationIT` include: 
∘ simplification achieved using predefined ports for keycloak and skills-service  `webEnvironment=SpringBootTest.WebEnvironment.DEFINED_PORT` and `keycloak.setPortBindings(["7779:8080"])`
∘ after using predefined ports all of the properties can be loaded on the startup which more accurately mimics production environment
∘ `userExists should return true when user exists` and `userExists should return false when user does not exist` simply mock up `userExists` method then the mocked version was utilized; instead assertions of `/userExists/{user}` endpoint were added to `SAML2 login flow with Keycloak integration` test; also asserting of `/app/userInfo` endpoint was added
∘ added db data reset on startup of the integration tests

Reply:
Made the improvements as suggested with static port